File: AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/data/main/ListItem.java
package com.applovin.enterprise.apps.demoapp.data.main;

/**
 * All the list items on main screen should confirm to this interface.
 * <p>
 * Created by santoshbagadi on 2019-09-10.
 */
public interface ListItem
{
    int TYPE_SECTION_HEADER = 0;
    int TYPE_AD_ITEM        = 1;
    int TYPE_FOOTER         = 2;

    /**
     * @return The type of list item.
     */
    int getType();
}

File: AppLovin MAX Demo App - Kotlin/app/src/main/java/com/applovin/enterprise/apps/demoapp/data/main/ListItem.kt
package com.applovin.enterprise.apps.demoapp.data.main

/**
 * Interface for all list items on the main screen.
 * <p>
 * Created by Harry Arakkal on 9/17/2019
 */
interface ListItem
{
    companion object
    {
        const val SECTION_HEADER = 0
        const val AD_ITEM = 1
        const val FOOTER = 2;
    }

    val type: Int
}
